### ISSUE TITLE
TextFields with filters renders incorrectly on hidpi screens

### ISSUE THREAD

**Issue by buble** [openfl/openfl#2565]
**Describe the bug**
After those commits 779455ac2 2d6d4ceaf texts are crisp on hidpi screens, however I see this issue with filters

Probably we need to improve filters rendering too?

**To Reproduce**
```
package;

import openfl.text.TextFieldType;
import openfl.text.TextField;
import openfl.filters.GlowFilter;
import openfl.events.FocusEvent;
import openfl.display.Sprite;

class Main extends Sprite
{
	public function new()
	{
		super();

		var textLabel = new TextField();
		textLabel.type = TextFieldType.INPUT;
		textLabel.wordWrap = true;
		textLabel.multiline = true;
		textLabel.textColor = 0x000000;
		textLabel.embedFonts = true;
		textLabel.text = "Type your text...";
		textLabel.border = true;
		addChild( textLabel );

		textLabel.addEventListener( FocusEvent.FOCUS_IN, function(e) {
			textLabel.filters = [
				new GlowFilter( 0x00FF00, 1, 4, 4, 8 )
			];
		} );

		textLabel.addEventListener( FocusEvent.FOCUS_OUT, function(e) {
			textLabel.filters = [];
		} );
	}
}
```

**Screenshots**

![text_bug](https://user-images.githubusercontent.com/6225437/171353643-76a4ba17-35ea-474a-899d-7a56b9428095.gif)

**OpenFL Targets**
I'm using latest git versions of lime and openfl 
macOS 11.6.6

**Comment by joshtynjala** [comment]
Confirmed. Interestingly, if the filter is applied immediately, instead of in an `FocusEvent.FOCUS_IN` listener, the text is rendered correctly.

```hx
// this works fine
var textLabel = new TextField();
textLabel.type = TextFieldType.INPUT;
textLabel.wordWrap = true;
textLabel.multiline = true;
textLabel.textColor = 0x000000;
textLabel.embedFonts = true;
textLabel.text = "Type your text...";
textLabel.border = true;
addChild( textLabel );
textLabel.filters = [
	new GlowFilter( 0x00FF00, 1, 4, 4, 8 )
];
```

**Comment by joshtynjala** [comment]
More info: Waiting for at least one full frame to render before setting the filters has a similar effect to waiting until `FocusEvent.FOCUS_IN`. This gives the same low resolution text.

```haxe
var textLabel = new TextField();
textLabel.type = TextFieldType.INPUT;
textLabel.wordWrap = true;
textLabel.multiline = true;
textLabel.textColor = 0x000000;
textLabel.embedFonts = true;
textLabel.text = "Type your text...";
textLabel.border = true;
addChild( textLabel );

var frameCount = 0;
textLabel.addEventListener(Event.ENTER_FRAME, function enterFrameHandler(e) {
	if (frameCount < 1) {
		frameCount++;
		return;
	}
	textLabel.removeEventListener(Event.ENTER_FRAME, enterFrameHandler);
	textLabel.filters = [new GlowFilter(0x00FF00, 1, 4, 4, 8)];
});
```

**Comment by joshtynjala** [comment]
Closed in commit f657ca664e9ccb8c6e87ebf264904255708113b0
